test/unit/mongo/MongoSearchProviderTest.php

```diff
@@ -598,6 +598,160 @@ public function testDeleteSearchDocumentsByTypeIdDoNotDeleteNonMatchingDocuments
+
+    public function testCountSearchDocumentsWithFilters()
+    {
+        $tripod = $this->getMockBuilder('\Tripod\Mongo\Driver')
+           ->setConstructorArgs(['CBD_testing', 'tripod_php_testing'])
+            ->getMock();
+
+        $filters = ['_cts' => ['$lte' => new \MongoDB\BSON\UTCDateTime(null)]];
+        $query = array_merge(['_id.type' => 'i_search_list'], $filters);
+        $collection = $this->getMockBuilder('\MongoDB\Collection')
+            ->disableOriginalConstructor()
+            ->setMethods(['count'])
+            ->getMock();
+        $search = $this->getMockBuilder('\Tripod\Mongo\MongoSearchProvider')
+            ->setMethods(['getCollectionForSearchSpec'])
+            ->setConstructorArgs([$tripod])
+            ->getMock();
+
+        $search->expects($this->once())
+            ->method('getCollectionForSearchSpec')
+            ->with('i_search_list')
+            ->will($this->returnValue($collection));
+
+        $collection->expects($this->once())
+            ->method('count')
+            ->with($query)
+            ->will($this->returnValue(89));
+
+        $this->assertEquals(89, $search->count('i_search_list', $filters));
+    }
+
+    public function testDeleteSearchDocumentsBySearchId()
+    {
+        $tripod = $this->getMockBuilder('\Tripod\Mongo\Driver')
+            ->setConstructorArgs(['CBD_testing', 'tripod_php_testing'])
+            ->getMock();
+
+        $collection = $this->getMockBuilder('\MongoDB\Collection')
+            ->disableOriginalConstructor()
+            ->setMethods(['deleteMany'])
+            ->getMock();
+
+        $deleteResult = $this->getMockBuilder('MongoDB\DeleteResult')
+            ->setMethods(['getDeletedCount'])
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $deleteResult->expects($this->once())
+            ->method('getDeletedCount')
+            ->will($this->returnValue(9));
+
+        $search = $this->getMockBuilder('\Tripod\Mongo\MongoSearchProvider')
+            ->setMethods(['getCollectionForSearchSpec', 'getSearchDocumentSpecification'])
+            ->setConstructorArgs([$tripod])
+            ->getMock();
+
+        $search->expects($this->once())
+            ->method('getSearchDocumentSpecification')
+            ->with('i_search_list')
+            ->will($this->returnValue(['_id' => 'i_search_list']));
+
+        $search->expects($this->once())
+            ->method('getCollectionForSearchSpec')
+            ->with('i_search_list')
+            ->will($this->returnValue($collection));
+
+        $collection->expects($this->once())
+            ->method('deleteMany')
+            ->with(['_id.type' => 'i_search_list'])
+            ->will($this->returnValue($deleteResult));
+
+        $this->assertEquals(9, $search->deleteSearchDocumentsByTypeId('i_search_list'));
+    }
+
+    public function testDeleteSearchDocumentsBySearchIdWithTimestamp()
+    {
+        $timestamp = new \MongoDB\BSON\UTCDateTime(null);
```

Comment:
do you need to pass null?

Reply:
Yes, although this is (maddeningly) PHP *version* dependent.  It's not necessary for 5.6+, but required for <= 5.5

Reply:
p.s. I don't know why

Reply:
p.p.s. But you'll notice it's in our util functions in the main apps, as well